gapis/server/server.go

```diff
@@ -129,13 +129,27 @@ func (s *server) CheckForUpdates(ctx context.Context, includePrereleases bool) (
 	}
 	var mostRecent *service.Release
 	mostRecentVersion := app.Version
+	mostRecentDevVersion := -1
+
 	for _, release := range releases {
 		if !includePrereleases && release.GetPrerelease() {
 			continue
 		}
+
+		// tagName is either:
+		// Regular release: v<major>.<minor>.<point>
+		// Dev pre-release: v<major>.<minor>.<point>-dev-<dev>
 		var version app.VersionSpec
-		fmt.Sscanf(release.GetTagName(), "v%d.%d.%d", &version.Major, &version.Minor, &version.Point)
-		if version.GreaterThan(mostRecentVersion) {
+		tagName := release.GetTagName()
+		devVersion := -1
+		numFields, err := fmt.Sscanf(tagName, "v%d.%d.%d-dev-%d", &version.Major, &version.Minor, &version.Point, &devVersion)
+		if err != nil && numFields < 3 {
+			// some non-release tags have other format, e.g. libinterceptor-v1.0
+			log.W(ctx, "Ignoring tag %s", tagName)
+			continue
+		}
+
+		if version.GreaterThan(mostRecentVersion) || (version.Equal(mostRecentVersion) && devVersion > mostRecentDevVersion) {
```

Comment:
Aren't dev releases going to be using the next version's major.minor? If that is the case, this is incorrect: If we currently are on M.N, and the dev release is M.(N+1), when we release M.(N+1), it will not be considered newer than the last M.(N+1) dev release.

Reply:
Fixed